mlflow/gemini/autolog.py

```diff
@@ -0,0 +1,46 @@
+import inspect
+
+import mlflow
+import mlflow.gemini
+from mlflow.entities import SpanType
+from mlflow.utils.autologging_utils.config import AutoLoggingConfig
+
+
+def _get_span_type(task) -> str:
+    import google.generativeai as genai
+
+    span_type_mapping = {
+        genai.GenerativeModel: SpanType.LLM,
+        genai.ChatSession: SpanType.CHAT_MODEL,
+    }
+    return span_type_mapping.get(task, SpanType.UNKNOWN)
+
+
+def construct_full_inputs(func, self, *args, **kwargs):
+    signature = inspect.signature(func)
+    arguments = signature.bind_partial(self, *args, **kwargs).arguments
+
+    if "self" in arguments:
+        arguments.pop("self")
+
+    if hasattr(self, "model_name"):
+        arguments["model_name"] = self.model_name
+
+    return arguments
+
+
+def patched_call(original, self, *args, **kwargs):
+    config = AutoLoggingConfig.init(flavor_name=mlflow.gemini.FLAVOR_NAME)
+
+    if config.log_traces:
+        with mlflow.start_span(
+            name=self.__class__.__name__,
```

Comment:
Can we add a method name to the span name, like `GenerativeModel.generate_content`? `GenerativeModel` have two methods to patch and Google may add more.

Reply:
I think we're only patching one method on `GenerativeModel` now, but agree that this can change in the future and having method name is safer!